ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/service/internal/RemoteFileServiceImpl.java

```diff
@@ -752,21 +890,49 @@ private SAMRecord filterMe(SAMRecord record, List<String> fields, List<String> t
         return record;
     }
 
+    /**
+     * Filters a variant context based on fields, tags, and excluded tags, if no
+     * fields, tags or notags are provided, the original context is returned.
+     *
+     * @param context - The context to be filtered
+     * @param fields - Fields to include after filtering
+     * @param tags - Tags to include after filtering.
+     * @param notags - Tags to exclude after filtering.
+     * @return The modified context
+     */
     private VariantContext filterMe(VariantContext context, List<String> fields, List<String> tags, List<String> notags) {
         // Default - leave record as it is
         if (fields == null && tags == null && notags == null) return context;
 
         Map<String, Object> attributes = context.getAttributes();
 
-        // TODO  
+        // TODO
 
         return context;
     }
 
+    /**
+     * Returns the size of the requested content, defined by the destination
+     * format, start coordinate and end coordinate. The format only effects the
+     * content size in the manner that AES encrypted data has a 16 octet random
+     * Initialization Vector (IV) which is discounted for decrypted files.
+     * See https://www.aescrypt.com/aes_file_format.html for details.
+     *
+     * @param reqFile - a File object, used to get the file size when no
+     *     coordinates are given.
+     * @param destinationFormat - File destination format, used to discount the
+     *     IV header size when set to 'plain'.
+     * @param startCoordinate - Start coordinate of content range, or 0.
+     * @param endCoordinate - End coordinate of content range, or 0.
+     * @return Size of the requested file content.
+     */
     private long getContentLength(File reqFile, String destinationFormat, long startCoordinate, long endCoordinate) {
         long length = 0;
 
         // EncryptionFormat
+        // The AES encrypted data has a 16 byte random IV at the start, so to
```

Comment:
This comment for the prefix is not correct. This `prefix` variable is to for destination file format length. If the user wants the `plain` format it means we don't need the header so we remove the prefix to 0 length. Otherwise, for AES destination file format it has to be 16 bits added before the file.